scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/FileSystemService.java

```diff
@@ -0,0 +1,167 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import edu.asu.diging.scriptoocloud.core.exceptions.DataFileStorageException;
+import edu.asu.diging.scriptoocloud.core.exceptions.DatasetStorageException;
+import edu.asu.diging.scriptoocloud.core.service.IFileSystemService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.*;
+import java.util.Objects;
+
+@Service
+@PropertySource("classpath:/config.properties")
+public class FileSystemService implements IFileSystemService {
+
+    @Value("${rootUploadLocation}")
+    private String rootLocationString;
+
+    @Autowired
+    public FileSystemService() {
+    }
+
+    /**
+     * Creates the dataset (and its required directories) on the filesystem.
+     *
+     * @param id       The id of the dataset.
+     * @param username The username of the dataset.
+     * @throws DatasetStorageException The exception thrown if the directory could not be created.
+     */
+    @Override
+    public void addDatasetDirectories(String id, String username) throws DatasetStorageException {
+        // First, the correct directory must be created
+        Path path = createPath(username, id);
+        File directory = path.toFile();
+        // if the Path was created successfully, create the directory
+        try {
+            if (!directory.exists()) {
+                Files.createDirectories(path);
+            }
+        } catch (IOException e) {
+            throw new DatasetStorageException("IOException occurred in addDatasetDirectories", e);
+        } catch (UnsupportedOperationException e) {
+            throw new DatasetStorageException("UnsupportedOperationException occurred in addDatasetDirectories", e);
+        } catch (SecurityException e) {
+            throw new DatasetStorageException("SecurityException occurred in addDatasetDirectories", e);
```

Comment:
these error message don't help anything, since they are too technical to be shown to the user but are also not telling someone looking at the logs anything new (that a securityexception occurred in the addDatasetDirectories method can be seen from the stack trace).